packages/gatsby/src/webpack/chunk-sorter.js

```diff
@@ -0,0 +1,80 @@
+/** https://github.com/jantimon/html-webpack-plugin/blob/master/lib/chunksorter.js */
+
+const toposort = require(`toposort`)
+const _ = require(`lodash`)
+
+/**
+  Sorts dependencies between chunks by their "parents" attribute.
+  This function sorts chunks based on their dependencies with each other.
+  The parent relation between chunks as generated by Webpack for each chunk
+  is used to define a directed (and hopefully acyclic) graph, which is then
+  topologically sorted in order to retrieve the correct order in which
+  chunks need to be embedded into HTML. A directed edge in this graph is
+  describing a "is parent of" relationship from a chunk to another (distinct)
+  chunk. Thus topological sorting orders chunks from bottom-layer chunks to
+  highest level chunks that use the lower-level chunks.
+*/
+module.exports = (chunks, chunkGroups) => {
```

Comment:
100% out of scope, but I wish we could create a utility containing this algo, and then have both us and html-webpack-plugin use it 🙃